packages/pluggableWidgets/accordion-native/src/Accordion.tsx

```diff
@@ -15,44 +16,51 @@ export function Accordion(props: Props): ReactElement | null {
     const [expandedGroups, setExpandedGroups] = useState<number[]>([]);
 
     if (!initialRender.current) {
+        initialRender.current = true;
         let initialExpandedGroups = props.groups.reduce(
-            (acc: number[], group: GroupsType, index: number): number[] =>
-                group.groupCollapsed === "groupStartExpanded" ? [...acc, index] : acc,
+            (acc, group, index): number[] =>
+                !props.collapsible || group.groupCollapsed === "groupStartExpanded" || expandedGroups.includes(index)
+                    ? [...acc, index]
+                    : acc,
             []
         );
-        initialExpandedGroups = !props.collapsible
-            ? initialExpandedGroups
-            : props.collapseBehavior === "singleExpanded"
-            ? [initialExpandedGroups[0]]
-            : initialExpandedGroups;
         setExpandedGroups(initialExpandedGroups);
-        initialRender.current = true;
     }
 
     const onPressGroupHeader = (group: GroupsType, index: number): void => {
         const expanded = expandedGroups.includes(index);
-        if (group.groupAttribute && isAvailable(group.groupAttribute)) {
-            group.groupAttribute?.setValue(!expanded);
+        if (group.groupCollapsedAttribute?.status === ValueStatus.Available) {
+            group.groupCollapsedAttribute.setValue(expanded);
         }
-        executeAction(group.groupOnChange);
         if (expanded) {
             collapseGroup(index);
         } else {
             expandGroup(index);
         }
+
+        executeAction(group.groupOnChange);
     };
 
     const collapseGroup = (index: number): void => {
-        setExpandedGroups(expandedGroups.filter(i => i !== index));
+        setExpandedGroups(oldArray => oldArray.filter(i => i !== index));
     };
     const expandGroup = (index: number): void => {
-        setExpandedGroups(props.collapseBehavior === "singleExpanded" ? [index] : [...expandedGroups, index]);
+        if (props.collapseBehavior === "singleExpanded") {
+            expandedGroups.map(i => {
+                if (index !== i && props.groups[i].groupCollapsedAttribute?.status === ValueStatus.Available) {
+                    props.groups[i].groupCollapsedAttribute?.setValue(true);
```

Comment:
It's probably an edge case, but with this check (line 51) inside the "is available" condition, and the "set state" call on line 54 outside the condition, you could get into a strange state. 

e.g.

1. JS engine reaches line 50, and the `groupCollapsedAttribute` changes status to `loading`
2. line 51 is skipped - so the attribute value remains whatever it was previously
3. line 54 is reached, collapsing all groups except the one tapped - getting data and widget appearance out of sync

Reply:
One solution, albeit maybe not the best, is to first check if all groups are available, then proceed